dask/highlevelgraph.py

```diff
@@ -67,7 +67,7 @@ def is_materialized(self) -> bool:
         """Return whether the layer is materialized or not"""
         return True
 
-    def get_output_keys(self) -> Set:
+    def get_output_keys(self) -> AbstractSet:
```

Comment:
- ``typing.AbstractSet`` and ``collections.abc.Set`` are immutable set-likes; ``frozenset`` and the output of ``dict.keys()`` are two implementations.
- ``typing.Set`` and ``collections.abc.MutableSet`` are mutable set-likes; ``set`` is one implementation.

Even when your implementation is actually returning a mutable object, it may be a good idea to declare it as immutable, particularly when it's exposing the internal state of your object, as doing so will cause mypy to raise an error if consumer code tries to write back into it. This is for example why I went with a generic Mapping (immutable) for HighLevelGraph.layers and dependencies, even if I could have declared them as dicts; letting the user write to them would potentially make them inconsistent with the memoized output of ``to_dict``.

``get_dependencies`` creates the output set on the fly every time, so it's OK to let the user write back to it. This is akin to the difference between dict.keys() in Python 2 vs. Python 3.

Analogously, when you accept a collection as an input parameter, it's always a good idea to declare it as an immutable abstract type, so that mypy will raise an error if you write back to it.
Case in point, ``cull`` writes back to its input. I did not spend time to investigate the performance and functional implications of working on a copy instead.